### PR TITLE
Clarify FacDB Mapped

### PR DESCRIPTION
**Pull request by SashaWeinstein**
# Fix "Change in fraction of mapped records by facdomain" section

This section lives on the "Review by classification level" page of the FacDB QAQC. On this page a user can review the change in total facilities by data source, facility type, facility subgroup, facility group, facility domain. 
There is also a section that was supposed to display similar stats for the change in fraction of mapped units. Before this PR the page had some issues. 

The biggest was that our QAQC on the FacDB side didn't count unmapped units from the previous release. This issue was addressed in [FacDB PR 550](https://github.com/NYCPlanning/db-facilities/pull/550).

The rest of the improvements were made on the QAQC streamlit side

- All stats displayed count records **with a geometry** instead of **without a geometry**
- The chart and tables were formatted to display percents with 2 decimal places and 
- A hover with raw counts was added
- Headers were made more clear
- An extra table was taken out so that the section on total counts and the section on mapped records follows the same pattern (one chart then one table)

Note that the change to the QAQC tables is a breaking change, so the new app will only work with QAQC tables produced after may 5th. If we want to keep the QAQC on previous versions working we need a workaround more complex than just merging here.

### CODE REVIEW

**Review by AmandaDoyle — COMMENTED** [review]
The one quick thing I want to flag is that I'm reluctant to merge into main for same reasons I gave in PLUTO QAQC.  Do we want have a master feature branch>

**Comment by SashaWeinstein** [comment]
I think a master feature branch for this repo makes a lot of sense, want to keep development going without changing what Lynn sees. Should we call this branch `dev`? I can make this change and re-open the PR once we feel good

File: src/facdb.py [hunk 1]
```diff
@@ -90,58 +91,108 @@ def count_comparison(df, width=1000, height=1000):
...
             """
-            Use the slide bar and drop down to change the difference
-            threshold and select the attribute to review
+            Use the number input bar to change the difference
+            threshold
             """
         )
-        st.header(f"Change in number of records by {level}")
+        st.header(f"Change in total number of records by {level}")
         st.write(f"diff > {thresh}")
```
Comment (td928) [inline src/facdb.py:194]: I think this description might be better. diff > 100 look weird to me when -100 also shows up. Absolute(diff) might be too incomprehensible also. How about "absolute change greater than 100"

File: setup.py [hunk 2]
```diff
@@ -7,9 +7,9 @@
     install_requires=[
         "sqlalchemy",
         "psycopg2-binary",
-        "pandas<1.1",
+        "pandas",
         "numpy",
-        "streamlit==1.8.1",
```
Comment (mbh329) [inline setup.py:12]: If we don't specify the version of the package here, it automatically installs the most recent version?

File: src/facdb.py [hunk 3]
```diff
@@ -90,58 +91,108 @@ def count_comparison(df, width=1000, height=1000):
...
             ["datasource", "factype", "facsubgrp", "facgroup", "facdomain"],
             index=0,
         )
+        st.sidebar.success(
+            """
+            Use the dropdown input bar to select an attribute to review
+            """
+        )
+        dff = qc_diff.groupby(level).sum()
+        thresh = st.sidebar.number_input(
```
Comment (mbh329) [inline src/facdb.py:179]: I like this feature a lot better than the slider. noice.

File: src/facdb.py (hunk 1 above)
Comment (SashaWeinstein) [reply src/facdb.py:194]: good comment, I will make this change now

File: src/facdb.py [hunk 4]
```diff
@@ -151,34 +202,43 @@ def by_classification():
...
-        dfff["pctwogeom_old"] = dfff["wogeom_old"] / dfff["count_old"]
-        dfff["pctwogeom_new"] = dfff["wogeom_new"] / dfff["count_new"]
-        dfff["pctwogeom_old"] = dfff["pctwogeom_old"].round(2)
-        dfff["pctwogeom_new"] = dfff["pctwogeom_new"].round(2)
-        dfff["diff"] = dfff["pctwogeom_new"] - dfff["pctwogeom_old"]
-        dfff["diff"] = dfff["diff"].round(2)
-        dfff = dfff.loc[(dfff["diff"] != 0) & (~dfff["diff"].isna()), :]
-        dfff = dfff.sort_values("diff")
+        # dfff.insert(0, level, dfff.index)
+        dfff["pct_mapped_old"] = dfff["with_geom_old"] / dfff["count_old"]
```
Comment (mbh329) [inline src/facdb.py:223]: Getting a key error for "with_geom_old" when trying to run `Change in fraction of mapped records by datasource`

**Comment by mbh329** [comment]
This PR doesn't impact this suggestion but I think the operator and oversight QC tables  under the general review could have more descriptive titles

File: setup.py (hunk 2 above)
Comment (SashaWeinstein) [reply setup.py:12]: yea that's correct

File: src/facdb.py (hunk 4 above)
Comment (SashaWeinstein) [reply src/facdb.py:223]: you need to point it to branch 553 for it to work

Comment (mbh329) [reply src/facdb.py:223]: sorry that was stupid on my end

**Review by mbh329 — APPROVED** [review]

**Pull request merged**